### PR TITLE
Add SQL support section for Hive connector

### PR DESCRIPTION
**Pull request by mosabua**
Only a first iteration for now. The whole Hive connector doc needs some restructuring. Imho this is good to go as is though .. or I could pull in more work from file types and a bunch of other clean up. Open to suggestions

### CODE REVIEW

File: docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -643,6 +643,48 @@ connector.
...
+:doc:`/sql/update` is only supported for transactional Hive tables with format
+ORC. ``UPDATE`` of partition or bucket columns is not supported.
+
+.. _hive-schema-table-management:
+
+Schema and table management
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector supports :ref:`sql-schema-table-management` with the exception of
+``ALTER SCHEMA``, since the Hive metastore does not support renaming schemas.
```
Comment (findepi) [inline docs/src/main/sphinx/connector/hive.rst:686]: is `ALTER SCHEMA ... SET AUTHORIZATION` not supported either?

**Comment by mosabua** [comment]
I think this is good to go now. 

Caveat - we say we do not support transaction (https://trino.io/docs/current/language/sql-support.html#transactions) and I assume that is correct. Follow up question from me would be.. where are transactions supported then? Anywhere?

File: docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -643,6 +643,40 @@ connector.
...
+* :ref:`sql-security-operations`, see also :ref:`hive-sql-standard-based-authorization`
+
+.. _hive-data-management:
+
+Data management
+^^^^^^^^^^^^^^^
+
+The :ref:`sql-data-management` functionality includes support for ``INSERT``,
+``UPDATE``, and ``DELETE`` statements in principle. The specifics depend on the
+used objects storage, file format, and metastore:
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:670]: Simplify and combine this sentence

File: docs/src/main/sphinx/connector/hive.rst [hunk 3]
```diff
@@ -643,6 +643,40 @@ connector.
...
+
+* :ref:`sql-security-operations`, see also :ref:`hive-sql-standard-based-authorization`
+
+.. _hive-data-management:
+
+Data management
+^^^^^^^^^^^^^^^
+
+The :ref:`sql-data-management` functionality includes support for ``INSERT``,
+``UPDATE``, and ``DELETE`` statements in principle. The specifics depend on the
```
Comment (electrum) [inline docs/src/main/sphinx/connector/hive.rst:669]: > ... includes support for ... statements, depending on the storage system, file format, and metastore:

File: docs/src/main/sphinx/connector/hive.rst [hunk 4]
```diff
@@ -999,19 +1033,8 @@ Drop a schema::
 
     DROP SCHEMA hive.web
 
-Hive connector limitations
---------------------------
-
-* :doc:`/sql/alter-schema` usage fails, since the Hive metastore does not support renaming schemas.
```
Comment (electrum) [inline docs/src/main/sphinx/connector/hive.rst:1005]: Should we mention this above?

File: docs/src/main/sphinx/connector/hive.rst (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/hive.rst:669]: updated

**Comment by mosabua** [comment]
Addressed all comments from @electrum and we discussed rest of content as well. Should be good to go now.

**Review by electrum — APPROVED** [review]

**Comment by electrum** [comment]
Thanks!

**Pull request merged**